lib/Dialect/TritonGPU/Transforms/OptimizeDotOperands.cpp

```diff
@@ -334,11 +332,13 @@ class TritonGPUOptimizeDotOperandsPass
     auto ret = pm.run(m);
 
     mlir::RewritePatternSet patterns(context);
-    patterns.add<SwizzleShmemConvert>(context);
+
     if (triton::gpu::TritonGPUDialect::getComputeCapability(m) >= 80)
       patterns.add<HoistLayoutConversion>(context);
     patterns.add<FuseTransHopper>(context);
+    patterns.add<SwizzleShmemConvert>(context);
     patterns.add<MMAV3UseRegOperand>(context);
+    ConvertLayoutOp::getCanonicalizationPatterns(patterns, context);
```

Comment:
Why this is necessary?

Reply:
I needed it because otherwise some convert were not removed before prefecth pass, I think that can be fixed by moving some of the canonicalizations for alloc/shared_load into folder. I'll see if I can get it done in this PR and remove that otherwise will add a comment.